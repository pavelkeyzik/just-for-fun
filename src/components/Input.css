.Input {
  font-size: 1em;
}
