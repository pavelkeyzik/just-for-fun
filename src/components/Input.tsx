import React from 'react';

import styles from './Input.module.css';

export function Input({ ...props }): JSX.Element {
  return <input data-testid="input" className={styles.input} {...props} />;
}
