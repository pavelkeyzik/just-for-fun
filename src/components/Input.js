import React from 'react';

import './Input.css';

export function Input({ ...props }) {
  return <input className="Input" {...props} />;
}
