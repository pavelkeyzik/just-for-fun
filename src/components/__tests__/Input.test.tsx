import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { Input } from '../Input';

test('Should be input element', () => {
  const { container } = render(<Input />);

  expect(container.firstChild).toMatchSnapshot('should be an input');
});

test('Should apply any given props to element', () => {
  const placeholder = 'Input placeholder';
  const { container } = render(<Input placeholder={placeholder} />);

  expect(container.firstChild).toMatchSnapshot(
    'should be an input with passed placeholder value',
  );
});

test('Should change value if user type text', () => {
  const newText = 'Typed something...';
  const { getByTestId } = render(<Input />);
  const inputElement = getByTestId('input') as HTMLInputElement;

  act(() => {
    fireEvent.change(inputElement, { target: { value: newText } });
  });

  expect(inputElement.value).toEqual(newText);
});
