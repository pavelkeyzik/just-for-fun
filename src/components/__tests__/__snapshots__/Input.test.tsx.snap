// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should apply any given props to element: should be an input with passed placeholder value 1`] = `
<input
  class="input"
  data-testid="input"
  placeholder="Input placeholder"
/>
`;

exports[`Should be input element: should be an input 1`] = `
<input
  class="input"
  data-testid="input"
/>
`;
